pkg/controller/lambda/function/setup.go

```diff
@@ -0,0 +1,499 @@
+			})
+		}
+	}
+	if !reflect.ValueOf(obj.Configuration.FileSystemConfigs).IsZero() {
+		awsFileSystemConfigs = obj.Configuration.FileSystemConfigs
+	}
+	// Check if counts match which should handle nil/empty
+	if len(fileSystemConfigs) != len(awsFileSystemConfigs) {
+		return false
+	}
+	// Check if actual values match, and only DeepEqual if there are actual values
+	if len(fileSystemConfigs) > 0 &&
+		!reflect.DeepEqual(fileSystemConfigs, awsFileSystemConfigs) {
+		return false
+	}
+	return true
+}
+
+func isUpToDateCode(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// GetFunctionOutput returns
+	// Code *FunctionCodeLocation `type:"structure"`
+	// which does not map to
+	// Code *FunctionCode `type:"structure" required:"true"`
+	// which is used when creating the function.
+	// We can't currently properly implement a comparison
+	return true
+}
+
+func isUpToDateLayers(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// The generator is creating layers with type of
+	// 	Layers []*string `json:"layers,omitempty"`
+	// Instead of []*svcsdk.Layer
+	// We can't properly implement a comparison until that is fixed
+	return true
+}
+
+func isUpToDateTracingConfig(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
+	mode := svcsdk.TracingModePassThrough
+	if !reflect.ValueOf(cr.Spec.ForProvider.TracingConfig).IsZero() &&
+		!reflect.ValueOf(cr.Spec.ForProvider.TracingConfig.Mode).IsZero() {
+		mode = *cr.Spec.ForProvider.TracingConfig.Mode
+	}
+	if !reflect.DeepEqual(mode, *obj.Configuration.TracingConfig.Mode) {
+		return false
+	}
+	return true
+}
+
+func isUpToDateSecurityGroupIDs(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
+	var securityGroupIDs []*string
+	var awsSecurityGroupIDs []*string
+	if !reflect.ValueOf(cr.Spec.ForProvider.VPCConfig).IsZero() &&
+		!reflect.ValueOf(cr.Spec.ForProvider.VPCConfig.SecurityGroupIDs).IsZero() {
+		securityGroupIDs = cr.Spec.ForProvider.VPCConfig.SecurityGroupIDs
+	}
+	if !reflect.ValueOf(obj.Configuration.VpcConfig).IsZero() &&
+		!reflect.ValueOf(obj.Configuration.VpcConfig.SecurityGroupIds).IsZero() {
+		awsSecurityGroupIDs = obj.Configuration.VpcConfig.SecurityGroupIds
+	}
+	// Check if counts match which should handle nil/empty
+	if len(securityGroupIDs) != len(awsSecurityGroupIDs) {
+		return false
+	}
+	// Check if actual values match, and only DeepEqual if there are actual values
+	if len(securityGroupIDs) > 0 &&
+		!reflect.DeepEqual(securityGroupIDs, awsSecurityGroupIDs) {
+		return false
+	}
+	return true
+}
+
+// returns which AWS Tags exist in the resource tags and which are outdated and should be removed
+func diffTags(spec map[string]*string, current map[string]*string) (map[string]*string, []*string) {
+	addMap := make(map[string]*string, len(spec))
+	removeTags := make([]*string, 0)
+	for k, v := range current {
+		if reflect.DeepEqual(spec[k], v) {
```

Comment:
```suggestion
		if aws.StringValue(spec[k]) == aws.StringValue(v) {
```

We don't need to use `reflect` for simple string comparisons.